### PR TITLE
Support for API Token authentication

### PR DESCRIPTION
**Pull request by mwpreston**
This PR adds support to connect to Rubrik clusters using API tokens rather than credential or Username/Password combinations.

## Description
A check is implemented before connecting to the Rubrik Cluster to detect whether Token or Credential based authentication was passed, and the proper parameters are passed.

## Motivation and Context
Many Rubrik customers have now implemented MFA on their clusters, which essentially removes the ability to connect to a rubrik cluster with the Powershell SDK using a Username/Password. This allows them to follow Rubrik best practices and generate API tokens for authentication with their automation scripts/toolsets.

## How Has This Been Tested?
This has been tested against Rubrik 5.0 and above and is working without issues.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.

### CODE REVIEW

File: Src/Public/Invoke-AsBuiltReport.Rubrik.CDM.ps1 [hunk 1]
```diff
@@ -19,6 +19,7 @@ function Invoke-AsBuiltReport.Rubrik.CDM {
     param (
         [String[]] $Target,
         [PSCredential] $Credential,
+        [String] $Token,
         [String]$StylePath
```
Comment (tpcarman) [inline Src/Public/Invoke-AsBuiltReport.Rubrik.CDM.ps1]: $StylePath should no longer be required.

**Review by tpcarman — CHANGES_REQUESTED** [review]
Remove `$StylePath` parameter and you should be good to go.

File: Src/Public/Invoke-AsBuiltReport.Rubrik.CDM.ps1 (hunk 1 above)
Comment (tpcarman) [reply Src/Public/Invoke-AsBuiltReport.Rubrik.CDM.ps1]: Will also need to update the `RequiredModules` in the module manifest to 
```
RequiredModules = @(
        @{
            ModuleName = 'AsBuiltReport.Core';
            ModuleVersion = '1.2.0'
        }
    )
```

**Comment by mwpreston** [comment]
@tpcarman removed StylePath, Updated RequiredModules  and Added Release.yml.  Do you mind having another quick look?

**Pull request merged**